### DISCUSSION TITLE
Where to place the current states animation? meta or context?

### DISCUSSION THREAD

**Discussion by pke** [statelyai/xstate#2644] [Q&A] [answered]
In my game state chart I want to play a win/lost animation.
I thought of defining the URI to the animationa each states "meta" info.

However its extremely difficult to extract the meta info during runtime from a nested machine, since the meta info is keyed by the current state name.

I'd like to get only the meta info for the current state, is this possible?

Or is driving the client via such meta-data discouraged and everything should be in "context" instead?

**Comment by ChrisShank** [comment, marked as answer]
Could you elaborate on what type of data you are putting in the `meta` field?

**Reply by pke** [reply]
Simplified version:

```ts
won: {
  on: {
     RESTART: "#game.lobby",
  },
  meta: {
    animation: "animations/won.mp4",
  }
}
```

But of course this kind of meta data is always static. It can't be returning a different animation based on the time or score the user achieved.
So its more and more clear, the animation to play during this state should be part of the context.

Different oppinions?

**Reply by ChrisShank** [reply]
Ya if its not 100% static then context is the way to go!

**Reply by ChrisShank** [reply]
Another alternative is that there is a mapping of the current state to an animation that loves outside of the machine. Might not suit your needs though!
